generator.go

```diff
@@ -88,15 +97,16 @@ func (s *TraceGenerator) generate_spans(ctx context.Context, fielder *Fielder, l
 		spancounts[rand.Intn(spansAtThisLevel)] += count
 	}
 
-	durationRemaining := time.Duration(rand.Intn(int(timeRemaining) / (nspans + 1)))
+	durationRemaining := time.Duration(rand.Intn(int(math.Ceil((float64(timeRemaining) / float64(nspans+1))))))
 	durationPerChild := (timeRemaining - durationRemaining) / time.Duration(spansAtThisLevel)
 
 	for i := 0; i < spansAtThisLevel; i++ {
 		durationThisSpan := durationRemaining / time.Duration(spansAtThisLevel-i)
 		durationRemaining -= durationThisSpan
 		time.Sleep(durationThisSpan / 2)
-		childctx, span := s.tracer.CreateSpan(ctx, fielder.GetServiceName(depth), level, fielder)
-		s.generate_spans(childctx, fielder, level+1, depth-1, spancounts[i]-1, durationPerChild)
+		childctx, span := s.tracer.CreateSpan(ctx, fielder.GetServiceName(i+1), level, fielder)
```

Comment:
The original intent was that service name was depth-dependent, and this breaks that. It's not wrong, but it does change the traces a lot. Maybe it doesn't matter but why did you feel the need to change this?

Reply:
Because now `depth` can be 0, the code panic when `depth` reaches 0

Reply:
I reverted this change